MEQP2/Sniffs/Classes/ConstructorOperationsSniff.md

```diff
@@ -0,0 +1,32 @@
+# Rule: Only DI operations are allowed in constructor
+## Background
+In Magento 2, the PHP constructor of a class is used to inject dependencies: Constructor-based Dependency Injection (DI).
+Because the constructor its purpose is to inject dependencies, other operations (or other logic) should not be part of the
+constructor.
+
+## Reasoning
+The constructor should only assign the values of constructor arguments to internal variables, like shown below:
+
+```php
+class Foo
+{
+    private $bar;
+
+    public function __construct(
+        Bar $bar
+    ) {
+        $this->bar = $bar;
+    }
+}
+```
+
+Any other logic than these dependency assignment operators should be excluded from the constructor. Having additional logic in
+the constructor makes it harder to debug and test the class.
```

Comment:
I'd add that you don't know in which stage of application initialization the class may already be instantiated, which can not only lead to hard-to-find bugs but also to performance issues.

Reply:
I was looking for a better phrasing of this but got stuck. Many eyes see more than 2. Thanks: https://github.com/jissereitsma/extdn-phpcs/commit/5093a755147a4ade6d68dbcf2d6f494cb347f581